src/index.js

```diff
@@ -19,29 +19,58 @@ function addColors(gradient, scale, colors) {
   }
 }
 
-const areaIsValid = (area) => area && area.right > area.left && area.bottom > area.top;
+function getScale(meta, axis) {
+  const key = axis + 'Scale'; 
+
+  if(meta.controller) {
```

Comment:
Both, v2 and v3 have `meta.controller`, so this in not a valid way to separate the versions.

`Chart.version` was introduced in v3, so could use that to separate.